### PR TITLE
allow specifying labels for job

### PR DESCRIPTION
**Pull request by yixingfu**
It is very hard to tell which job is which on AI platform with job-id information.  Allowing passing labels would be very helpful. 
Labels are dictionaries of strings prescribed in https://cloud.google.com/ai-platform/training/docs/resource-labels
Validation functions are added to check labels.  By default no labels are passed, where a suggestion message is shown when validating.

### CODE REVIEW

File: tensorflow_cloud/run.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def run(
     entry_point_args=None,
     stream_logs=False,
     docker_image_bucket_name=None,
+    labels = {},
```
Comment (pavithrasv) [inline tensorflow_cloud/run.py:42]: We will need to change the name to reflect what it is. `labels` is vague. May be `job_labels`?

Comment (pavithrasv) [inline tensorflow_cloud/run.py:42]: Please add docs for the new API param.

File: tensorflow_cloud/validate.py [hunk 2]
```diff
@@ -150,6 +153,53 @@ def _validate_cluster_config(chief_config, worker_count, worker_config):
...
         )
 
+def _validate_labels(labels):
+    """Validates labels conform guidelines at 
+       https://cloud.google.com/ai-platform/training/docs/resource-labels"""
+    if not labels:
+        print("No label for the job. Consider creating labels"
+              "for easier organizing submitted jobs.")
+
+    if len(labels) > 64:
```
Comment (pavithrasv) [inline tensorflow_cloud/validate.py:163]: AI platform specific label validation should live in gcp.py. We can just invoke that function from here.

File: tensorflow_cloud/validate.py [hunk 3]
```diff
@@ -150,6 +153,53 @@ def _validate_cluster_config(chief_config, worker_count, worker_config):
             "Received {}.".format(worker_count)
         )
 
+def _validate_labels(labels):
+    """Validates labels conform guidelines at 
+       https://cloud.google.com/ai-platform/training/docs/resource-labels"""
+    if not labels:
+        print("No label for the job. Consider creating labels"
```
Comment (pavithrasv) [inline tensorflow_cloud/validate.py:160]: Please reword this as "No labels provided for the training job. Please consider creating labels to help with retrieval of job information. Please see https://cloud.google.com/ai-platform/training/docs/resource-labels for more information."

+ All of this should live in gcp.py and invoked from here.

File: tensorflow_cloud/validate.py [hunk 4]
```diff
@@ -150,6 +153,53 @@ def _validate_cluster_config(chief_config, worker_count, worker_config):
...
+
+    if len(labels) > 64:
+        raise ValueError("Invalid labels: too many labels."
+                         "Expecting at most 64 labels."
+                         "Received {}.".format(len(labels)))
+
+    for k in labels:
+        v = labels[k]
+        if not k or not k[0].islower():
+            raise ValueError("Invalid labels:"
```
Comment (pavithrasv) [inline tensorflow_cloud/validate.py:171]: Please modify all message here to say 'Invalid job labels:'

File: tests/tensorflow_cloud/validate_test.py [hunk 5]
```diff
@@ -302,3 +317,123 @@ def test_invalid_tpu_accelerator_count(self):
...
+                requirements_txt="tests/testdata/requirements.txt",
+                chief_config=machine_config.COMMON_MACHINE_CONFIGS["K80_1X"],
+                worker_config=machine_config.COMMON_MACHINE_CONFIGS["K80_1X"],
+                worker_count=1,
+                region="us-central1",
+                args=None,
+                stream_logs=True,
+                docker_image_bucket_name=None,
+                called_from_notebook=False,
+                labels={'':''}
```
Comment (pavithrasv) [inline tests/tensorflow_cloud/validate_test.py:336]: Can we remove most params for which default values are sufficient for testing from here and the calls below?

File: tensorflow_cloud/validate.py [hunk 6]
```diff
@@ -34,6 +35,7 @@ def validate(
     stream_logs,
     docker_image_bucket_name,
     called_from_notebook,
+    labels={}
```
Comment (pavithrasv) [inline tensorflow_cloud/validate.py:38]: Please update docs.

File: tensorflow_cloud/deploy.py [hunk 7]
```diff
@@ -114,6 +117,8 @@ def _create_request_dict(
             the general workers in a distribution cluster.
         entry_point_args: Command line arguments to pass to the
             `entry_point` program.
+    Optional:
+        labels: Dict of strings for labels to organize jobs.
```
Comment (pavithrasv) [inline tensorflow_cloud/deploy.py:122]: for => of. Please add link to AI platform labels page here in docs.

File: tensorflow_cloud/run.py (hunk 1 above)
Comment (yixingfu) [reply tensorflow_cloud/run.py:42]: Sounds good. Changed all `labels` -> `job_labels`

File: tensorflow_cloud/validate.py (hunk 2 above)
Comment (yixingfu) [reply tensorflow_cloud/validate.py:163]: Thanks for pointing out. Moved.

File: tests/tensorflow_cloud/validate_test.py (hunk 5 above)
Comment (yixingfu) [reply tests/tensorflow_cloud/validate_test.py:336]: The validate function does not have default values and are all positional args. 

Instead I only left test for `job_labels` in valid args in validate_test.py, and moved others to gcp_test.py to specifically test the `validate_job_labels` function.

Comment (pavithrasv) [reply tests/tensorflow_cloud/validate_test.py]: Got it, Thank you!

File: tensorflow_cloud/deploy.py [hunk 8]
```diff
@@ -114,6 +117,9 @@ def _create_request_dict(
             the general workers in a distribution cluster.
         entry_point_args: Command line arguments to pass to the
             `entry_point` program.
+    Optional:
```
Comment (pavithrasv) [inline tensorflow_cloud/deploy.py:120]: Please remove optional header.
